src/main/java/uk/gov/hmcts/reform/pip/account/management/database/UserRepository.java

```diff
@@ -22,4 +22,6 @@ List<PiUser> findExistingByProvenanceId(@Param("provUserId") String provenanceUs
     List<PiUser> findVerifiedUsersByLastVerifiedDate(@Param("daysAgo") int daysSinceLastVerified);
 
     Optional<PiUser> findByEmail(String email);
+
+    Optional<PiUser> findByProvenanceUserId(String provenanceUserId);
```

Comment:
This will cause an issue if multiple IDAM's have the same provenance ID.

I think we either need to use the user ID instead or pass in both the provenance and the IDAM type to this method

Reply:
Done :)